core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -604,6 +612,10 @@ private Type handleResolvedField(Expression node, ResolvedField resolvedField, S
         @Override
         protected Type visitDereferenceExpression(DereferenceExpression node, StackableAstVisitorContext<Context> context)
         {
+            if (isAllRowsReference(node)) {
+                throw semanticException(NOT_SUPPORTED, node, "all rows labeled reference is not allowed in this context");
```

Comment:
The term "all rows labeled reference" is not intuitive. I'd rephase as `<identifier>.* not allowed in this context`